source/simulator/core.cc

```diff
@@ -506,9 +495,18 @@ namespace aspect
         //use it, we can run into problems with geometry_model->depth().
         AssertThrow ( parameters.pressure_normalization == "no",
                       ExcMessage("The free surface scheme can only be used with no pressure normalization") );
-        free_surface.reset( new FreeSurfaceHandler( *this, prm ) );
+
+        //Allocate the FreeSurfaceHandler object
+        free_surface.reset( new FreeSurfaceHandler<dim>( *this, prm ) );
       }
 
+    //The mapping is given by a degree four MappingQ for the case
+    //of a curved mesh, and a degree one MappingQ for a rectangular mesh.
+    //If a free surface is enabled, this is swapped out for a MappingQ1Eulerian,
+    //which allows for mesh deformation.
+    mapping.reset( new MappingQ<dim>( geometry_model->has_curved_elements() ? 4 : 1,
+                                      geometry_model->has_curved_elements() ? true : false ) );
```

Comment:
Oh, I don't think I understand. Why not leave this block of code in the member initializer list around line 187? If you did that, you could just do the replacement of the mapping in the constructor of `FreeSurfaceHandler` right away.